test/parallel/test-fs-glob.mjs

```diff
@@ -427,6 +427,7 @@ const pattern2 = [
     [`${absDir}/*{a,q}*`, './a/*{c,b}*/*'],
     [`${absDir}/foo`, 'a/c', ...(common.isWindows ? [] : ['a/symlink/a/b/c'])],
   ],
+  [ 'a/**', () => true, [] ],
```

Comment:
Considering this:
```suggestion
  [ 'a/**', () => true, [] ],
  [ 'a/**', [ '*' ], [] ], // passes
  [ 'a/**', [ '**' ], [] ], // passes
  [ 'a/**', [ 'a/**' ], [] ], // not passes (expected [] got ['a'])
```
I would expect `globSync` to return an empty array whenever `exclude === [ pattern ]`, and i don't think `a/**` glob should match `a/` itself in the first place.

Reply:
Adding more test cases is a fantastic way to provide a code review. Thanks! I'll see if I can carve out some time to fix this.

Reply:
OK, I think I've fixed this.